File: c++/153. Find Minimum in Rotated Sorted Array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int ls = nums.size(), l, r, m;
        l = 0;
        r = ls - 1;
        if (nums[l] <= nums[r])
            return nums[l];
        while (r - l > 1) {
            m = l + (r-l>>1);
            if (nums[m] > nums[l])
                l = m;
            else
                r = m;
        }
        
        return nums[r];
    }
};

File: python/153. Find Minimum in Rotated Sorted Array.py
class Solution(object):
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        l, r = 0, len(nums)-1
        while l < r:
            m = (l+r)>>1
            k = nums[m]
            if k < nums[r]: # m->r递增
                r = m
            else:
                l = m + 1
        return nums[l]